themes/theme-default/src/client/composables/useSidebarItems.ts

```diff
@@ -195,11 +195,14 @@ export const resolveMultiSidebarItems = (
   const sidebarPath = resolveLocalePath(sidebarConfig, path)
   const matchedSidebarConfig = sidebarConfig[sidebarPath] ?? []
 
+  if (matchedSidebarConfig === 'heading') {
+    return resolveAutoSidebarItems(page, sidebarDepth)
+  }
   return resolveArraySidebarItems(
     page,
     router,
     path,
-    matchedSidebarConfig,
+    matchedSidebarConfig as SidebarConfigArray,
```

Comment:
Hi, `matchedSidebarConfig ` should be type-narrowed in line 198, what's the infer type here that you have to use a type cast?

Reply:
This is left over from when I had added a helper function to check if the config was `heading`/`auto` and the then the TS isn't able to determine that you already have determined that `matchedSidebarConfig` must be a `SidebarConfigArray`.  This ended up being the spot it was needed so I removed the helper function and now, you are right, the explicit type case is not longer needed.